### PR TITLE
fix(API): always return `null` when no receipt was found

### LINKED ISSUE #1412: Analyze the failed test (Contract: DoubleEndedQueue) in workflows/openzeppelin_test_16_19.yml
https://github.com/axonweb3/axon/actions/workflows/openzeppelin_test_16_19.yml 的 Contract: DoubleEndedQueue failed to run twice:

### ProviderError: Custom error: can not get receipt

1. https://github.com/axonweb3/axon/actions/runs/6093863030/job/16534301433#step:15:91
2. https://github.com/axonweb3/axon/actions/runs/6093672541/job/16533737625#step:15:140

### PR DESCRIPTION
**Pull request by KaoImin**
## What this PR does / why we need it?

This PR refactor the process of `eth_getReceiptByHash` API. The new process is:
1. Try to get from storage
	a. If return is `Some`, return
2. Otherwise, return `Ok(None)`

### What is the impact of this PR?

No Breaking Change

**PR relation**:
- Fix #1412

<details><summary>CI Settings</summary><br/>

### **CI Usage**

**Tip**: Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [x] E2E Tests
- [x] Web3 Compatible Tests
- [x] OCT 1-5 And 12-15
- [x] OCT 6-10
- [x] OCT 11
- [x] OCT 16-19
- [x] v3 Core Tests

### **CI Description**

| CI Name                                   | Description                                                               |
| ----------------------------------------- | ------------------------------------------------------------------------- |
| *Chaos CI*                                | Test the liveness and robustness of Axon under terrible network condition |
| *Cargo Clippy*                            | Run `cargo clippy --all --all-targets --all-features`                     |
| *Coverage Test*                           | Get the unit test coverage report                                         |
| *E2E Test*                                | Run end-to-end test to check interfaces                                   |
| *Code Format*                             | Run `cargo +nightly fmt --all -- --check` and `cargo sort -gwc`           |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                                       |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3                        |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin                      |

</details>

### CODE REVIEW

**Review by sunchengzhu — DISMISSED** [review]

**Review by driftluo — DISMISSED** [review]

**Review by sunchengzhu — DISMISSED** [review]

File: core/api/src/jsonrpc/impl/web3.rs [hunk 1]
```diff
@@ -606,29 +606,32 @@ impl<Adapter: APIAdapter + 'static> Web3RpcServer for Web3RpcImpl<Adapter> {
...
         }
+
+        if self.adapter.mempool_contains_tx(ctx, &hash).await {
+            return Ok(None);
+        }
+
+        Err(Error::Custom(format!(
+            "Can not get receipt by hash {:?}",
+            hash
+        )))
```
Comment (Flouse) [inline core/api/src/jsonrpc/impl/web3.rs]: Suppose 0x85d995eba9763907fdf35cd2034144dd9d53ce32cbec21349d4b12823c6860c1 is a nonexistent transaction hash.
And the result from `eth_getTransactionReceipt` is null, not Error.
```bash
curl --location --request POST 'https://rpc.flashbots.net/' \
--header 'Content-Type: application/json' \
--header 'Accept: */*' \
--data-raw '{
    "id": 42,
    "jsonrpc": "2.0",
    "method": "eth_getTransactionReceipt",
    "params": ["0x85d995eba9763907fdf35cd2034144dd9d53ce32cbec21349d4b12823c6860c1"]
}'

# Result
{"id":42,"result":null,"jsonrpc":"2.0"}

```

**Note**: rpc.flashbots.net is [Ethereum Mainnet](https://chainlist.org/chain/1)'s RPC.

Comment (KaoImin) [reply core/api/src/jsonrpc/impl/web3.rs]: Does this mean query any nonexistent transaction hash should return `Ok(None)`?

Comment (Flouse) [reply core/api/src/jsonrpc/impl/web3.rs]: I think it should return a transaction receipt object, or `null` when no receipt was found.

See https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_gettransactionreceipt

**Review by Flouse — APPROVED** [review]

**Review by chaoticlonghair — APPROVED** [review]

**Pull request merged**